packages/react-components/react-list-preview/stories/List/MultipleActionsWithPrimary.stories.tsx

```diff
@@ -0,0 +1,182 @@
+import {
+  Button,
+  Caption1,
+  makeResetStyles,
+  makeStyles,
+  Menu,
+  MenuItem,
+  MenuList,
+  MenuPopover,
+  MenuTrigger,
+  mergeClasses,
+  shorthands,
+  Text,
+  tokens,
+} from '@fluentui/react-components';
+import { MoreHorizontal20Regular } from '@fluentui/react-icons';
+import { List, ListItem, ListProps } from '@fluentui/react-list-preview';
+
+import * as React from 'react';
+
+const useListItemRootStyles = makeResetStyles({
+  position: 'relative',
+  flexGrow: '1',
+  gap: '8px',
+  border: '1px solid grey',
+  alignItems: 'center',
+  borderRadius: '8px',
+  gridTemplate: `"preview preview preview" auto
+      "header action secondary_action" auto / 1fr auto auto
+    `,
+});
+
+const useStyles = makeStyles({
+  list: {
+    display: 'flex',
+    flexDirection: 'column',
+    ...shorthands.gap('16px'),
+    maxWidth: '300px',
+  },
+  listItem: {
+    display: 'grid',
+    ...shorthands.padding('8px'),
+  },
+  caption: {
+    color: tokens.colorNeutralForeground3,
+  },
+  image: {
+    maxWidth: '100%',
+    ...shorthands.borderRadius('5px'),
+  },
+  title: {
+    fontWeight: 600,
+    display: 'block',
+  },
+  preview: {
+    ...shorthands.gridArea('preview'),
+    ...shorthands.overflow('hidden'),
+  },
+  header: { ...shorthands.gridArea('header') },
+  action: { ...shorthands.gridArea('action') },
+  secondaryAction: { ...shorthands.gridArea('secondary_action') },
+});
+
+const CustomListItem = (props: { title: string; value: string }) => {
+  const listItemStyles = useListItemRootStyles();
+  const styles = useStyles();
+  const { value } = props;
+
+  // This will be triggered by user pressing Enter or clicking on the list item
+  const onAction = React.useCallback(event => {
+    // This prevents the change in selection on click/Enter
+    event.preventDefault();
+    alert(`Triggered custom action!`);
+  }, []);
+
+  return (
+    <ListItem
+      value={props.value}
+      className={mergeClasses(listItemStyles, styles.listItem)}
+      aria-label={value}
+      onAction={onAction}
+    >
+      <div role="gridcell" className={styles.preview}>
+        <img className={styles.image} src={`https://picsum.photos/seed/${value}/300/130/`} alt="Presentation Preview" />
```

Comment:
- Let's use `Image` component there
- Let's not use third-party CDNs as we don't know what it will host in future, you can use images from `packages/fluentui/docs/src/public/images/wireframe` (available as https://fabricweb.azureedge.net/fabric-website/assets/images/wireframe/image.png) or upload yours (see `packages/fluentui/docs/README.md`)